src/App.tsx

```diff
@@ -274,8 +284,13 @@ function WorkerTrigger ({
       <label>
         <span>Search</span>
         <input autoFocus={true} type="text" placeholder={placeholders[placeholderIndex]} value={searchCriteria} ref={searchInputRef} onChange={(event) => {
-          navigate('/')
-          setSearchCriteria(event.target.value)
+          const value = event.target.value
+          setSearchCriteria(value)
+          const searchLocation = {
+            pathname: '/',
+            search: value ? new URLSearchParams({ s: value }).toString() : undefined
+          }
+          navigate(searchLocation, { replace: location.pathname === '/' })
```

Comment:
This `replace` parameter makes the router use `history.replaceState()` instead of `history.pushState()` and prevents typing on the search field from creating lots of entries on the browser history :)